### PR TITLE
Allow extensions to be able to make requests to the typescript extension's tsserver via commands 

### PR DESCRIPTION
**Pull request by orta**
This PR relates to https://github.com/microsoft/TypeScript-Website/issues/2156 and https://github.com/microsoft/vscode/pull/138211

This allows other extensions to make calls to the TypeScript Language Server, saving the need to create a second set of parallel tsservice-like infra in extensions like the typescript playground.

---

Tested by an extensions which calls:

```ts
console.log("Sending")
vscode.commands.executeCommand('typescript.tsserverRequest', 'emit-output', { file: "/Users/ortatherox/dev/js/berry/packages/plugin-git/sources/GitResolver.ts" } ).then(r => {
	console.log("Sent")
	console.log(r)
})
```

Which correctly returns the response to the request on the other side:

Screen Shot 2021-12-01 at 4 37 37 PM

### CODE REVIEW

**Comment by orta** [comment]
( Fails look unrelated )

File: extensions/typescript-language-features/src/commands/tsserverRequests.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+	public constructor(
+		private readonly lazyClientHost: Lazy<TypeScriptServiceClientHost>
+	) { }
+
+	public execute(requestID: keyof TypeScriptRequests, args?: any, config?: any) {
+		// A cancellation token cannot be passed through the command infrastructure
+		const token = nulToken;
+		try {
+			return this.lazyClientHost.value.serviceClient.execute(requestID, args, token, config);
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/commands/tsserverRequests.ts:23]: For the error handling, should we be awaiting the result of this call? Not sure when the code will throw as it is currently written

**Review by mjbvz — COMMENTED** [review]
Implementation looks good. I'm a little concerned about exposing this though since it gives extensions low level control and could make it difficult to understand why the server is in an odd state

What specific commands do you need to run for the playground?

**Comment by orta** [comment]
Updated based on a chat we had yesterday. 

I've removed my custom `try catch` error handling, at this point all the commands are reads and I don't think it should be possible to knock over the tsserver with them.

I don't know all of the commands the playground needs, but realistically it's probably a subset of the amount I've added here. It's enough to provide pretty solid introspection tooling.

**Comment by mjbvz** [comment]
Thanks! LGTM

**Pull request merged**